chore: publish unicorn and registry1 flavors of runtime

decleaver opened the pull request:
## Description

Removes artifacts from releases and publishes "unicorn" and "registry1" flavors of runtime

## Related Issue

- #509

TristanHoladay commented:
LGTM! only thing i thought of was maybe making the IB login an action. I'm pretty sure renovate will catch the image tag in the registry1 values file as well, but we should just be on the lookout for that in case it doesn't.

Also, noting we should maybe run nightly tests for each of the flavors.

decleaver commented:
> LGTM! only thing i thought of was maybe making the IB login an action. I'm pretty sure renovate will catch the image tag in the registry1 values file as well, but we should just be on the lookout for that in case it doesn't.
> 
> Also, noting we should maybe run nightly tests for each of the flavors.

Not sure about renovate.... i'll look into it some more and i can update the nightly tests too, to test both flavors

decleaver commented:
Added smoke tests for the different flavors. Wanted to run e2e tests, but we can run into issues where the released flavor version doesn't contain updates needed to run the current e2e tests.

UncleGedd commented:
For the tasks:
1) `smoke` and `smoke-flavor`
2) `install-runtime-in-cluster` and `install-flavored-runtime-in-cluster`

Instead of having flavor-specific tasks can we combine these into a single one?

UncleGedd commented on zarf.yaml:
```diff
@@ -25,3 +27,18 @@ components:
+        namespace: uds-runtime
+        version: 0.1.0
+        valuesFiles:
+          - values/registry1-values.yaml
```
You can also DRY this up a bit with the pattern over in uds-core, [example](https://github.com/defenseunicorns/uds-core/commit/0011852dd6c8f1305e2fa0c837db45f3c1801c31)

decleaver commented:
> For the tasks:
> 
> 1. `smoke` and `smoke-flavor`
> 2. `install-runtime-in-cluster` and `install-flavored-runtime-in-cluster`
> 
> Instead of having flavor-specific tasks can we combine these into a single one?

As is, this lets us run the smoke test and each of the flavored smoke tests in parallel in our workflows (speeding things up). This also lets us pass down the flavor var from the top level in the workflow without having to set that var in additional tasks (https://github.com/defenseunicorns/uds-runtime/pull/519/commits/a79696e69d861629082851c46a742477d63ce7ce). More than happy to combine those tasks though if preferred.

decleaver commented:
> > For the tasks:
> > 
> > 1. `smoke` and `smoke-flavor`
> > 2. `install-runtime-in-cluster` and `install-flavored-runtime-in-cluster`
> > 
> > Instead of having flavor-specific tasks can we combine these into a single one?
> 
> As is, this lets us run the smoke test and each of the flavored smoke tests in parallel in our workflows (speeding things up). This also lets us pass down the flavor var from the top level in the workflow without having to set that var in additional tasks ([a79696e](https://github.com/defenseunicorns/uds-runtime/commit/a79696e69d861629082851c46a742477d63ce7ce)). More than happy to combine those tasks though if preferred.

updated based on convo, adding new workflow that runs once a week or manually as desired. Refactored configuration with common zarfconfig.

UncleGedd commented on .github/workflows/flavor-tests.yaml:
```diff
@@ -0,0 +1,36 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+# This workflow runs smoke tests for the unicorn and registry1 flavors of UDS Runtime.
+name: Flavor Tests
```
How about post-release-tests or something to that effect? IMO `Flavor Tests` doesn't indicate that these are released artifacts (update file name as well)

UncleGedd commented on hack/flavors/common/zarf.yaml:
```diff
@@ -0,0 +1,25 @@
+# Copyright 2024 Defense Unicorns
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+kind: ZarfPackageConfig
```
```suggestion
# Copyright 2024 Defense Unicorns
# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial

kind: ZarfPackageConfig
```

UncleGedd reviewed (dismissed):
Tested out the Zarf pkg locally, also asked @TristanHoladay to double check the YAML to try to avoid any release-day thrash
